navigation/index.tsx

```diff
@@ -39,11 +47,19 @@ export default function AppNavigator() {
             headerTitleStyle: { color: theme.colors.white },
           }}
           initialRouteName="Home">
-          <Stack.Screen
-            name="Home"
-            options={{ headerShown: false }}
-            component={HomeScreen}
-          />
+          {hasFriends ? (
+            <Stack.Screen
+              name="Home"
+              options={{ headerShown: false }}
+              component={HomeScreen}
+            />
+          ) : (
+            <Stack.Screen
+              name="NoFriends"
+              options={{ headerShown: false }}
+              component={NoFriendsScreen}
+            />
+          )}
```

Comment:
we don't need an extra screen to render as the initial route
this `NoFriendsScreen.tsx` should render inside `Home.tsx`